Makefile

```diff
@@ -306,8 +306,8 @@ KUSTOMIZE_VERSION ?= v5.5.0 #(dprince: bumped to aquire new features like --load
 CONTROLLER_TOOLS_VERSION ?= v0.14.0
 CRD_MARKDOWN_VERSION ?= v0.0.3
 KUTTL_VERSION ?= 0.17.0
-GOTOOLCHAIN_VERSION ?= go1.21.0
-OC_VERSION ?= 4.14.0
+GOTOOLCHAIN_VERSION ?= go1.22.0
+OC_VERSION ?= 4.16.0
```

Comment:
4.18 not yet available

```
wget https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.18.0/openshift-client-linux-4.18.0.tar.gz -O - |\
tar xz)
--2025-04-23 08:46:54--  https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.18.0/openshift-client-linux-4.18.0.tar.gz
Resolving mirror.openshift.com (mirror.openshift.com)... 3.167.56.39, 3.167.56.96, 3.167.56.87, ...
Connecting to mirror.openshift.com (mirror.openshift.com)|3.167.56.39|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2025-04-23 08:46:54 ERROR 404: Not Found.
```

Reply:
Can't we just leave OC_VERSION at 4.16 here? We do still support running on that version...

Reply:
it was 4.14, I bumped it to 4.16. but I think I'll move this to a separate PR. its not really related to the golang bump.

Reply:
moved to https://github.com/openstack-k8s-operators/openstack-operator/pull/1431